Build failure: python311.pkgs.grip

### Steps To Reproduce

Steps to reproduce the behavior:
1. include python3.pkgs.grip in configuration
2. Checkout unstable to:

```
"nixpkgs": {
       "locked": {
       "lastModified": 1707689078,
        "narHash": "sha256-UUGmRa84ZJHpGZ1WZEBEUOzaPOWG8LZ0yPg1pdDF/yM=",
        "owner": "NixOS",
        "repo": "nixpkgs",
        "rev": "f9d39fb9aff0efee4a3d5f4a6d7c17701d38a1d8",
        "type": "github"
  }
```

### Build log

```
Running phase: installCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.11.7, pytest-7.4.4, pluggy-1.4.0
rootdir: /build/source
configfile: pytest.ini
collecting ... ^Mcollected 22 items / 6 deselected / 16 selected                                

tests/test_api.py ..........F

=================================== FAILURES ===================================
___________________________________ test_app ___________________________________

monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7ffff51b77d0>
tmpdir = local('/build/pytest-of-nixbld/pytest-0/test_app0')

    def test_app(monkeypatch, tmpdir):
        monkeypatch.setenv('GRIPHOME', str(tmpdir))
        zero_path = input_filename('zero.md')
        zero_output = output_file('app', 'zero.html')
        gfm_test_path = input_filename('gfm-test.md')
        gfm_test_output = output_file('app', 'gfm-test.html')
        assets = GitHubAssetManagerMock()
    
        with GitHubRequestsMock() as responses:
>           assert Grip(zero_path, assets=assets).render() == zero_output

tests/test_api.py:260: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Grip 'grip.app'>, route = '/'

    def render(self, route=None):
        """
        Renders the application and returns the HTML unicode that would
        normally appear when visiting in the browser.
        """
        if route is None:
            route = '/'
        with self.test_client() as c:
            response = c.get(route, follow_redirects=True)
>           encoding = response.charset
E           AttributeError: 'WrapperTestResponse' object has no attribute 'charset'

grip/app.py:400: AttributeError
=========================== short test summary info ============================
FAILED tests/test_api.py::test_app - AttributeError: 'WrapperTestResponse' object has no attribute 'charset'
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
================== 1 failed, 10 passed, 6 deselected in 0.27s ==================
```

### Additional context

Add any other context about the problem here.

### Notify maintainers

@mweinelt 

### Metadata

```
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.4, NixOS, 24.05 (Uakari), 24.05.20240207.f8e2ebd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/store/snp5f95xhkz89kdw9703kana3m6xfd0a-source`
```

--- a/pkgs/development/python-modules/grip/default.nix
+++ b/pkgs/development/python-modules/grip/default.nix
@@ -1,5 +1,6 @@
 { lib
 , fetchFromGitHub
+, fetchpatch
 # Python bits:
 , buildPythonPackage
 , pytest
@@ -24,6 +25,15 @@ buildPythonPackage rec {
     rev = "v${version}";
     hash = "sha256-CHL2dy0H/i0pLo653F7aUHFvZHTeZA6jC/rwn1KrEW4=";
   };
+
+  patches = [
+    # https://github.com/NixOS/nixpkgs/issues/288478
+    (fetchpatch {
+      name = "set-default-encoding.patch";
+      url = "https://github.com/joeyespo/grip/commit/2784eb2c1515f1cdb1554d049d48b3bff0f42085.patch";
+      hash = "sha256-veVJKJtt8mP1jmseRD7pNR3JgIxX1alYHyQok/rBpiQ=";
+    })
+  ];
 
   nativeCheckInputs = [ pytest responses ];
 

